### ISSUE TITLE
Markers Overlapping

### ISSUE THREAD

**Issue by ziadagh** [appoly/ARCore-Location#61]
For some reason, my markers keep colliding and are displayed in the same location with the same distance, even though they are set with different coordinates.
What could be the problem?

Here is a code snippet of my work:

```
 markers.add(new Marker(55.2875768847111, 25.2264987732707, "DWTC 1", "View"));
 markers.add(new Marker(55.2874789395990, 25.2265584635088, "DWTC 3", "View"));
```

```
CompletableFuture<ViewRenderable> markerLayout = ViewRenderable.builder().setView(context, R.layout.example_layout_1).build();
CompletableFuture.allOf(markerLayout).handle((notUsed, throwable) -> {
    if (throwable != null) {
        DemoUtils.displayError(context, "Unable to load view renderables.", throwable);
        return null;
    }
    try {
        ViewRenderable vr = markerLayout.get();
        Node n = getView(vr, marker);
        n.setName(marker.getName());
        LocationMarker locationMarker = new LocationMarker(marker.getLongitude(), marker.getLatitude(), n);
        locationMarker.setRenderEvent(node -> {
            View eView = vr.getView();
            TextView distanceTextView = eView.findViewById(R.id.markerDistance);
            String distance = Math.round(node.getDistanceInAR()) + " m";
            distanceTextView.setText(distance);
            eView.setOnTouchListener((v, event) -> {
                GeoDbSingleton.toast(marker.getName() + " was touched.");
                return false;
            });
        });
        checkIfFoundAndReplace(locationScene, locationMarker);
    } 
    catch (InterruptedException | ExecutionException ex) {
        DemoUtils.displayError(context, "Unable to load renderables.", ex);
    }
    return null;
});
```

```
private Node getView(ViewRenderable vr, Marker marker) {
    Node base = new Node();
    base.setRenderable(vr);
    TextView title = vr.getView().findViewById(R.id.markerName);
    title.setText(marker.getName());
    return base;
}
```

```
 private void checkIfFoundAndReplace(LocationScene locationScene, LocationMarker locationMarker) {
    if(locationScene.mLocationMarkers.size() >= 2){
      for(int i=0; i<locationScene.mLocationMarkers.size(); i++){
        if(locationScene.mLocationMarkers.get(i).node.getName().equals(locationMarker.node.getName())){
            locationScene.mLocationMarkers.get(i).anchorNode.getAnchor().detach();
            locationScene.mLocationMarkers.get(i).anchorNode.setAnchor(null);
            locationScene.mLocationMarkers.get(i).anchorNode.setEnabled(false);
            locationScene.mLocationMarkers.get(i).anchorNode = null;
            locationScene.mLocationMarkers.remove(i);

            locationScene.mLocationMarkers.add(locationMarker);
            locationScene.refreshAnchors();
            return;
        }
      }
    }else {
         locationScene.mLocationMarkers.add(locationMarker);
    }
  }
```

**Comment by johnwedgbury** [comment]
Try swapping the lat/lng values?

**Comment by ziadagh** [comment]
For me it worked by just swapping the lat and long values, I had entered them wrong so the location I was trying to show was too far away from me so they appeared to be overlapping.
Once I switched them I could see both markers clearly.
If you have the same issue, try the same fix, and set the markers at a close distance from you and move towards them to confirm that they are not overlapping up close. Unless you are working at a very small distance for the two from each other.
